aten/src/ATen/native/cpu/Vec256.h

```diff
@@ -121,6 +136,70 @@ class Vec256<float> {
   inline void operator=(const Vec256<float>& b) {
     values = b.values;
   }
+  inline Vec256<float> map(float (*f)(float)) {
+    __at_align32__ float tmp[8];
+    store(tmp);
+    for (size_t i = 0; i < 8; i++)
+      tmp[i] = f(tmp[i]);
+    Vec256<float> ret;
+    ret.load(tmp);
+    return ret;
+  }
+  inline Vec256<float> abs() {
+    Vec256<float> ret;
+    __m256 mask = _mm256_set1_ps(-0.f);
+    ret.values = _mm256_andnot_ps(values, mask);
+    return ret;
+  }
+
+#ifdef __AVX2__
+  inline Vec256<float> exp() {
+    Vec256<float> ret;
+    ret.values = exp256_ps(values);
+    return ret;
+  }
+#else
+  inline Vec256<float> exp() {
+    return map(std::exp);
+  }
+#endif
+
+#ifdef __AVX2__
+  inline Vec256<float> log() {
+    Vec256<float> ret;
+    ret.values = log256_ps(values);
```

Comment:
Do those really require AVX2? IIRC they worked fine with just AVX

Reply:
Interestingly enough I get tons of strange errors if I use __AVX__ instead of __AVX2__. We also guard it with AVX2 within TH.

Reply:
It uses integer AVX2 instructions. The library includes some code to emulate them with SSE instructions, but it'll require some small modifications to the header. (Mostly removing `#warning`)

I don't think we have many users with AVX but not AVX2, so I'm not too worried about the perf difference right now.